CONTRIBUTING.md

```diff
@@ -1,248 +1,271 @@
+- **Configuration Settings**: GUC settings to customize the extension's behavior (e.g., API keys, batch sizes).
+- **Triggers and Scheduling**: Defines triggers and cron jobs for automated data processing.
+- **Embedding and Search Operations**: Integrates with external AI services to generate embeddings and enables vector search capabilities.
 
-:wrench: When progressing through these steps, refer to the following for troubleshooting:
+**When to Contribute Here**:
 
-```bash
-cat ~/.pgrx/15.log
-```
+- Modifying database-specific logic or background job management.
+- Working on PostgreSQL interactions with external embedding services.
+- Enhancing in-database vector search functionalities.
 
-#### 3.1. Apply configurations
+### `core` Directory
 
-Prior to compiling and running `pg_vectorize`, it's essential to update the `postgresql.conf` file.
-`pgrx` uses a Postgres version-specific data directory, each containing its own `postgresql.conf` file.
-The following example, utilizes Postgres version 15.
-If you're using a different version, please alter the file path value `data-<postgres-version>` and run the following:
+Contains core logic and abstractions that support vectorization and embedding functionalities.
 
-```bash
-<your-editor> ~/.pgrx/data-15/postgresql.conf
-```
+**Key Components**:
 
-Within this document, add the following:
+- **Providers**: API clients and wrappers for external AI/embedding services.
+- **Data Structures**: Definitions of embeddings and related types.
+- **Embedding Processing**: Logic for handling embedding requests and responses.
 
-```text
-shared_preload_libraries = 'pg_cron, vectorize'
-cron.database_name = 'postgres'
-vectorize.embedding_service_url = 'http://localhost:3000/v1/embeddings'
-```
+**When to Contribute Here**:
 
-:wrench: Note: If your machine is running a MacOS, you may need to apply the following configurations to Cargo's config file:
+- Adding support for new AI/embedding providers.
+- Modifying embedding generation logic.
+- Implementing enhancements that are platform-independent.
 
-```
-<your-editor> ~/.cargo/config
-```
+## Setting Up Your Development Environment
 
-```text
-[target.'cfg(target_os="macos")']
-# Postgres symbols won't be available until runtime
-rustflags = ["-Clink-arg=-Wl,-undefined,dynamic_lookup"]
+### 1. Initialize PGRX
+
+The `cargo pgrx init` command initializes the PGRX development environment by downloading and compiling the required PostgreSQL versions.
+
+Run the following command:
+
+```bash
+cargo pgrx init
 ```
 
-#### 3.2. Clone and enter directory
+**MacOS Users**:
+
+- If you're on MacOS, you might need to configure Cargo for dynamic linking due to Postgres symbols not being available until runtime.
+
+  ```bash
+  mkdir -p ~/.cargo
+  echo '[target.'cfg(target_os="macos")']' >> ~/.cargo/config.toml
+  echo 'rustflags = ["-Clink-arg=-Wl,-undefined,dynamic_lookup"]' >> ~/.cargo/config.toml
+  ```
```

Comment:
If there's an issue w/ the config.toml then we should update the project's config file. But from what I can tell its already the same? Perhaps there is an issue with the project's config file getting picked up by the cli.

https://github.com/tembo-io/pg_vectorize/blob/main/extension/.cargo/config.toml